python/pyspark/sql/types.py

```diff
@@ -586,7 +586,10 @@ def fromInternal(self, micros: int) -> Optional[datetime.timedelta]:
 
 
 class YearMonthIntervalType(AnsiIntervalType):
-    """YearMonthIntervalType, represents year-month intervals of the SQL standard"""
+    """YearMonthIntervalType, represents year-month intervals of the SQL standard
+
+    Note that this data type doesn't support collection: df.collect/take/head.
```

Comment:
Let's use `Note` section in numpydoc (https://numpydoc.readthedocs.io/en/latest/format.html#notes)